src/MapWindow/GlueMapWindowItems.cpp

```diff
@@ -67,27 +76,31 @@ GlueMapWindow::ShowMapItems(const GeoPoint &location,
   if (task)
     builder.AddTaskOZs(*task);
 
-  const Airspaces *airspace_database = airspace_renderer.GetAirspaces();
-  if (airspace_database)
-    builder.AddVisibleAirspace(*airspace_database,
-                               airspace_renderer.GetWarningManager(),
-                               computer_settings.airspace,
-                               settings.airspace, basic,
-                               calculated);
+  if(settings.item_list.add_airspace) {
+    const Airspaces *airspace_database = airspace_renderer.GetAirspaces();
+    if (airspace_database)
+      builder.AddVisibleAirspace(*airspace_database,
+                                 airspace_renderer.GetWarningManager(),
+                                 computer_settings.airspace,
+                                 settings.airspace, basic,
+                                 calculated);
+  }
 
   if (visible_projection.GetMapScale() <= 4000)
     builder.AddThermals(calculated.thermal_locator, basic, calculated);
 
+  // flarm-traffic could be more interested than waypoints...if no interest, disable it
+  if(settings.item_list.add_traffic)
+    builder.AddTraffic(basic.flarm.traffic);
+
   if (waypoints)
-    builder.AddWaypoints(*waypoints);
+    builder.AddWaypoints(*waypoints,only_landable);
 
 #ifdef HAVE_NOAA
   if (noaa_store)
     builder.AddWeatherStations(*noaa_store);
 #endif
 
-  builder.AddTraffic(basic.flarm.traffic);
```

Comment:
This changes the order of items. Why?

Reply:
it is moved to then beginning, flarm-traffic could be more interested during flights, Only 32 mapitems are possible, if you use a great range, there is no change for traffic-items, if waypoints filled the list.
if no interest or to much traffic, you can disable traffic-items